docs/gene_sets.md

```diff
@@ -0,0 +1,136 @@
+
+
+
+Example:
+
+| gene_set_name | gene_set_description | gene_symbol | gene_description | provenance1      | provenance1_description |
+|---------------|----------------------|-------------|------------------|------------------|-------------------------|
+| club.cell     | description          | CCKAR       | description      | Pubmed ID XYZ123 | Primary Pubmed ID       |
+| club.cell     |                      | SCGB3A2     | description      | Pubmed ID ABC456 | Primary Pubmed ID       |
+| club.cell     |                      | CYP2F2      | description      | Pubmed ID DCF678 | Primary Pubmed ID       |
+| macrophage    | description          | CD68        | description      |                  |                         |
+| macrophage    |                      | CD163       | description      |                  |                         |
+
+
+## Mandatory Header
+
+The first row MUST contain header columns using reserved names in the following order:
+
+* `gene_set_name`
+* `gene_set_description`
+* `gene_symbol`
+* `gene_description`
+
+Publishers MAY include additional header columns. It is RECOMMENDED that these custom columns observe the same self-documenting style - `column_name` and `column_description` for easier comprehension by data consumers.
+
+## Rows
+
+Each subsequent row describes a gene in a gene set.
+<br><br>
+
+### Values
+
+The values for `gene_set_name`, `gene_set_description`, and `gene_symbol` MUST NOT contain illegal ASCII characters or sequences. If the following cases are detected, validation MUST display a warning and fail the upload:
+
+* control characters (decimal 0-31)
+* DEL (decimal 127)
+* leading spaces (decimal 32) in a field - "     This is an example"
+* trailing spaces (decimal 32) in a field - "This is an example     " 
+* multiple spaces (decimal 32) "internal" to a field - "This     is an example"
+
+### `gene_set_name`
+
+---
+
+The `gene_set_name` column MUST contain a value.
+
+If the `gene_set_name` is missing, validation MUST display a warning and fail the upload. This is illustrated by **~~?~~** in the example:
+
+| gene_set_name | gene_set_description | gene_symbol | gene_description |
+|---------------|----------------------|-------------|------------------|
+| club.cell     | description          | CCKAR       | description      |
+|     **~~?~~** |                      | SCGB3A2     | description      |
+| club.cell     |                      | CYP2F2      | description      |
+
+
+All `gene_symbol(s)` for a `gene_set_name` MUST be on contiguous rows; otherwise, an out-of-order row is interpreted as a duplicate `gene_set_name`. In this case, validation MUST display a warning and fail the upload. This is illustrated by **~~club.cell~~** in the example:
```

Comment:
As Brian noted, I think the contiguity requirement is onerous and should be removed.  It causes as many problems as it solves.

Reply:
Pablo and you have convinced me. We will remove the requirement from the portal and this document.